src/run-button.ts

```diff
@@ -0,0 +1,216 @@
+  }
+
+  private updateRunButtonVisibility(): void {
+    if (!this._runButton) {
+      return;
+    }
+
+    // Show run button on hover or if cell hasn't been executed
+    if (this._isHovered || !this.executionCount) {
+      this._runButton.show();
+      this._promptIndicator.hide();
+    } else {
+      this._runButton.hide();
+      this._promptIndicator.show();
+    }
+  }
+}
+
+export class JENotebookContentFactory extends Notebook.ContentFactory {
+  createInputPrompt(): JEInputPrompt {
+    return new JEInputPrompt();
+  }
+
+  createNotebook(options: Notebook.IOptions): Notebook {
+    return new Notebook(options);
+  }
+}
+
+/**
+ * Plugin that provides the custom notebook factory with run buttons
+ */
+export const notebookFactoryPlugin: JupyterFrontEndPlugin<NotebookPanel.IContentFactory> = {
+  id: 'jupytereverywhere:notebook-factory',
+  description: 'Provides notebook cell factory with input prompts',
+  provides: NotebookPanel.IContentFactory,
+  requires: [IEditorServices],
+  autoStart: true,
+  activate: (app: JupyterFrontEnd, editorServices: IEditorServices) => {
+    const editorFactory = editorServices.factoryService.newInlineEditor;
+
+    const factory = new JENotebookContentFactory({
+      editorFactory
+    });
+
+    return factory;
+  }
+};
+
+/**
+ * Plugin that sets up run buttons on notebook cells
+ */
+export const runCellButtonPlugin: JupyterFrontEndPlugin<void> = {
+  id: 'jupytereverywhere:run-cell-button',
+  description: 'Add run buttons to notebook cells',
+  autoStart: true,
+  requires: [INotebookTracker],
+  optional: [ITranslator],
+  activate: (app: JupyterFrontEnd, tracker: INotebookTracker, translator?: ITranslator) => {
+    const trans = (translator ?? nullTranslator).load('jupyterlab');
+
+    const runButtonFactory = (panel: NotebookPanel) =>
+      new ToolbarButton({
+        icon: EverywhereIcons.run,
+        onClick: () => {
+          void NotebookActions.runAndAdvance(panel.content, panel.sessionContext);
+        },
+        tooltip: trans.__('Run this cell and advance')
+      });
+
+    tracker.widgetAdded.connect((_, panel) => {
+      const cellListChanged = () => {
+        panel.content.widgets.forEach(cell => {
+          cell.ready
+            .then(() => {
+              if (cell.inputArea) {
+                const prompt = (cell.inputArea as any).prompt || (cell.inputArea as any)._prompt;
+
+                if (prompt && 'runButton' in prompt) {
+                  prompt.runButton = runButtonFactory(panel);
+                }
+              }
+            })
+            .catch(() => {
+              // no-op
+            });
+        });
+      };
+
+      panel.content.model?.cells.changed.connect(cellListChanged);
+
+      // Also run immediately for existing cells
+      cellListChanged();
+
+      panel.disposed.connect(() => {
+        panel.content.model?.cells.changed.disconnect(cellListChanged);
+      });
+    });
+  }
+};
```

Comment:
I would remove this and instead just invoke the `notebook:run-cell` command directly in the prompt area. This looks just way to complex for what is needed to me.

```suggestion
```